kahuna/public/js/components/gr-downloader/gr-downloader.js

```diff
@@ -55,37 +55,40 @@ downloader.controller('DownloaderCtrl', [
         );
 
         downloads$.subscribe((zip) => {
-            const file = zip.generate({ type: 'uint8array' });
-            const blob = new Blob([file], { type: 'application/zip' });
-
-            const isTooBig = blob.size > maxBlobSize;
-
-            const createDownload = () => {
-                const url = $window.URL.createObjectURL(blob);
-                $window.location = url;
-            };
-
-            const refuseDownload = () => {
-                const maxSize = bytesToSize(maxBlobSize);
-                const blobSize = bytesToSize(blob.size);
-
-                // the things we do for a happy linter...
-                const message = [
-                    'Download is too big!',
-                    `Max file size: ${maxSize}. Your download: ${blobSize}.`,
-                    'Consider downloading smaller batches.'
-                ].join('\n');
-
-                $window.alert(message);
-            };
-
-            if (isTooBig) {
-                refuseDownload();
-            } else {
-                createDownload();
-            }
-
-            ctrl.downloading = false;
+            zip.generateAsync({ type: 'uint8array' }).then(file => {
```

Comment:
was this synchronous (i.e. blocking) before?
(or is that irrelevant given its in a callback anyway?)

Reply:
probably was blocking?